test/time_and_test_refactor.jl

```diff
@@ -87,7 +88,7 @@ for N = [1,10]
 
     dx0   = zeros(2N)
     Random.seed!(42)
-    #display(@benchmark(nd!($dx0, randn(2*$N), ($ω, 5.), 0.)))
+    display(@benchmark($nd_anti!($dx0, randn(2*$N), ($ω, 5.), 0.)))
```

Comment:
You are benchmarking the allocating `randn`, try
```julia
    x0 = rand(2N)
    display(@benchmark($nd!($dx0, $x0, ($ω, 5.), 0.)))
```
(i've also added the `$` in front of `nd!` otherwise it uses the `nd!` from the global state which is not present if you run the script in a clean julia session)

Reply:
Does this really get you to 0 allocations? For me these changes are a matter of 1 or 2 allocations

Reply:
yes sorry i was checking the `benchmarking/time_and_test,jl` file....